samples/ssmlAddresses.js

```diff
@@ -0,0 +1,128 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+// [START tts_ssml_address_imports]
+// Imports the Google Cloud client library
+const textToSpeech = require('@google-cloud/text-to-speech');
+
+// Import other required libraries
+const fs = require('fs');
+//const escape = require('escape-html');
+const util = require('util');
+// [END tts_ssml_address_imports]
+
+// [START tts_ssml_address_audio]
+/**
+ * Generates synthetic audio from a String of SSML text.
+ *
+ * Given a string of SSML text and an output file name, this function
+ * calls the Text-to-Speech API. The API returns a synthetic audio
+ * version of the text, formatted according to the SSML commands. This
+ * function saves the synthetic audio to the designated output file.
+ *
+ * ARGS
+ * ssmlText: String of tagged SSML text
+ * outfile: String name of file under which to save audio output
+ * RETURNS
+ * nothing
+ *
+ */
+async function ssmlToAudio(ssmlText, outFile) {
+  // Creates a client
+  const client = new textToSpeech.TextToSpeechClient();
+
+  // Constructs the request
+  const request = {
+    // Select the text to synthesize
+    input: {ssml: ssmlText},
+    // Select the language and SSML Voice Gender (optional)
+    voice: {languageCode: 'en-US', ssmlGender: 'MALE'},
+    // Select the type of audio encoding
+    audioConfig: {audioEncoding: 'MP3'},
+  };
+
+  // Performs the Text-to-Speech request
+  const [response] = await client.synthesizeSpeech(request);
+  // Write the binary audio content to a local file
+  const writeFile = util.promisify(fs.writeFile);
+  await writeFile(outFile, response.audioContent, 'binary');
+  console.log('Audio content written to file ' + outFile);
+}
+// [END tts_ssml_address_audio]
+
+// [START tts_ssml_address_ssml]
+/**
+ * Generates SSML text from plaintext.
+ *
+ * Given an input filename, this function converts the contents of the input text file
+ * into a String of tagged SSML text. This function formats the SSML String so that,
+ * when synthesized, the synthetic audio will pause for two seconds between each line
+ * of the text file. This function also handles special text characters which might
+ * interfere with SSML commands.
+ *
+ * ARGS
+ * inputfile: String name of plaintext file
+ * RETURNS
+ * a String of SSML text based on plaintext input
+ *
+ */
+function textToSsml(inputFile) {
+  let rawLines = '';
+  // Read input file
+  try {
+    rawLines = fs.readFileSync(inputFile, 'utf8');
+  } catch (e) {
+    console.log('Error:', e.stack);
+    return;
+  }
+
+  // Replace special characters with HTML Ampersand Character Codes
```

Comment:
I'd consider putting all of this character manipulation into its own function

Reply:
In my [Python](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/texttospeech/ssml_addresses/tts.py) and [Java](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/texttospeech/cloud-client/src/main/java/com/example/texttospeech/SsmlAddresses.java) implementation of this method, I use the html `escape` method. The Node.js equivalent is `html.escapers`, but the linter tells me that it's deprecated. There is an `escape` method native to Node.js, but it replaces all spaces with '%20', which the `html.escape` methods for Python and Java don't. Because these samples are for a [tutorial](https://cloud.google.com/text-to-speech/docs/ssml-tutorial), I want to be consistent, which is why I hard-coded the character manipulation here. I want to be somewhat consistent with the Python and Java versions of this sample (or is that a misguided instinct?), which is why I'm nervous about making a new escape method. Any recommendations on the best way to proceed.

Reply:
That's a really good point that I actually don't know the answer to. @beccasaurus what's the policy on writing a helper function that's not in the canonical - like here when the escape method is deprecated in JS?